compiler/rustc_mir_transform/src/pass_manager.rs

```diff
@@ -90,6 +90,11 @@ pub(super) trait MirPass<'tcx> {
     fn is_mir_dump_enabled(&self) -> bool {
         true
     }
+
+    /// Returns `true` if this pass must be run; should be disabled for passes
+    /// are strictly optimizations. If this is `false`, `optimize(none)` will
+    /// disable the pass.
```

Comment:
```suggestion
    /// Returns `true` if this pass must be run (i.e. it is required for soundness).
    /// For passes which are strictly optimizations, this should return `false`.
    /// If this is `false`, `#[optimize(none)]` will disable the pass.
```

3 changes:
- Add an explanation of "must"
- Fix the sentence "should be disabled for passes are strictly optimizations" (I'm not sure what it means for a function to be disabled and there is a missing word...)
- Make it clear that `optimize(none)` is an attribute